modules/usage/module.php

```diff
@@ -0,0 +1,439 @@
+ *
+ * Elementor usage module handler class is responsible for registering and
+ * managing Elementor usage modules.
+ *
+ */
+class Module extends BaseModule {
+	const META_KEY = '_elementor_elements_usage';
+	const OPTION_NAME = 'elementor_elements_usage';
+
+	/**
+	 * @var bool
+	 */
+	private $is_document_saving = false;
+
+	/**
+	 * Get module name.
+	 *
+	 * Retrieve the usage module name.
+	 *
+	 * @access public
+	 *
+	 * @return string Module name.
+	 */
+	public function get_name() {
+		return 'usage';
+	}
+
+	/**
+	 * Get formatted usage
+	 *
+	 * Retrieve formatted usage, for frontend.
+	 *
+	 * @return array
+	 */
+	public function get_formatted_usage() {
+		$usage = [];
+
+		foreach ( get_option( self::OPTION_NAME, [] ) as $doc_type => $elements ) {
+			$doc_class = Plugin::$instance->documents->get_document_type( $doc_type );
+
+			if ( $doc_class ) {
+				$doc_title = $doc_class::get_title();
+			} else {
+				$doc_title = $doc_type;
+			}
+
+			$tab_group = $doc_class::get_property( 'admin_tab_group' );
+
+			if ( $tab_group ) {
+				$doc_title = ucwords( $tab_group ) . ' - ' . $doc_title;
+			}
+
+			// Sort by key.
+			ksort( $elements );
+
+			foreach ( $elements as $element_type => $data ) {
+				unset( $elements[ $element_type ] );
+
+				if ( in_array( $element_type, [ 'section', 'column' ], true ) ) {
+					continue;
+				}
+
+				$widget_instance = Plugin::$instance->widgets_manager->get_widget_types( $element_type );
+
+				if ( $widget_instance ) {
+					$widget_title = $widget_instance->get_title();
+				} else {
+					$widget_title = $element_type;
+				}
+
+				$elements[ $widget_title ] = $data['count'];
+			}
+
+			$usage[ $doc_type ] = [
+				'title' => $doc_title,
+				'elements' => $elements,
+			];
+		}
+
+		return $usage;
+	}
+
+	/***
+	 * Set saving flag
+	 *
+	 * Called on elementor/document/before_save
+	 *
+	 * @param Document $document
+	 */
+	public function before_document_save( $document ) {
+		$this->remove_from_global( $document );
+
+		$this->is_document_saving = true;
+	}
+
+	/**
+	 * Save the current usage
+	 *
+	 * Called on elementor/document/after_save
+	 *
+	 * @param Document $document
+	 */
+	public function after_document_save( $document ) {
+		if ( DB::STATUS_PUBLISH === $document->get_post()->post_status ) {
+			$this->save_document_usage( $document );
+		}
+
+		$this->is_document_saving = false;
+	}
+
+	/**
+	 * On Status Change
+	 *
+	 * Called on transition_post_status
+	 *
+	 * @param string $new_status
+	 * @param string $old_status
+	 * @param WP_Post $post
+	 */
+	public function on_status_change( $new_status, $old_status, $post ) {
+		// If it's from elementor editor, the usage should be saved via `before_document_save`/`after_document_save`.
+		if ( $this->is_document_saving ) {
+			return;
+		}
+
+		$document = Plugin::$instance->documents->get( $post->ID );
+
+		if ( ! $document ) {
+			return;
+		}
+
+		$is_update = 'publish' === $old_status && 'publish' === $new_status;
```

Comment:
Remove - It's handheld by `before_document_save`

Reply:
```suggestion
		$is_update = 'publish' === $old_status && 'publish' === $new_status;
```